File: 17th-May-Trace_Path/python/solution.py

#This Code is written by Bhaskar


class Solution:
    def isPossible(self, n, m, s):
        # code here
        i=0 
        j=0
        extR=0 
        extL=0
        extT=0
        extB=0
        for ind in range (0 , len(s)):
            ch = s[ind]
            if(ch=='L'):
                j = j-1
                extL = min(extL,j)
            elif(ch=='R'):
                j = j+1
                extR = max(extR,j)
            elif(ch=='U'):
                i = i+1
                extT = max(extT,i)
            elif(ch=='D'):
                i = i-1
                extB = min(extB,i)
        Horizontal = abs(extL) + abs(extR)
        Vertical = abs(extT) + abs(extB)
        
        if(Horizontal<m and Vertical<n):
            return 1;
        return 0;

File: 17th-May-Trace_Path/c++/solution.cpp
/*
This Code is written by Bhaskar
*/

class Solution{
public:
    int isPossible(int n, int m, string s){
        // code here
        int i=0 , j=0; //lets assume
        int extR = 0 ,extL=0 , extT=0,extB=0;
        for(int ind=0;ind<s.length();ind++){
            char ch = s[ind];
            switch (ch){
                case 'L' :  j--;
                            extL = min(extL,j);
                    break;
                case 'R' :  j++;
                            extR = max(extR,j);
                    break;
                case 'D' :  i--;
                            extB = min(extB,i);
                    break;
                case 'U' :  i++;
                            extT = max(extT,i);
                    break;
            }

        }
        int lenH = abs(extR) + abs(extL);
        int lenV = abs(extT) + abs(extB);
        
        if(lenH < m && lenV <n) return 1;
        return 0;
    }
};
